client/src/components/History/CurrentHistory/HistoryPanel.vue

```diff
@@ -80,7 +80,7 @@ const querySelectionBreak = ref(false);
 const dragTarget = ref<EventTarget | null>(null);
 const contentItemRefs = computed(() => {
     return historyItems.value.reduce((acc: ContentItemRef, item) => {
-        acc[`item-${item.id}`] = ref(null);
+        acc[`item-${item.hid}`] = ref(null);
```

Comment:
Neither id nor hid are unique. If uniqueness is an advantage (it very probably is) you can create a composite key of item-type (`hda | hdca | dce`) and id.

Reply:
That's a good idea, doing that

Reply:
Actually, wouldn't `item.type_id` also work? Or is that not necessarily unique?

Reply:
type_id would work if we consistently serialized it, but it's better if we compose that on the client (that's just what we do on the backend)

Reply:
I see.
I used `{history_content_type}-{id}` instead:
https://github.com/galaxyproject/galaxy/pull/17648/commits/1aa96aa175b7b94a868a5154988d8d5cff98f0c4